File: introduction/ITP1_7_A-Grading.c
#include <stdio.h>

int main() {
  int m, f, r;
  while(1) {
    scanf("%d %d %d", &m, &f, &r);
    if(m + f + r == -3) break;
    char grade = 'F';
    if(m != -1 && f != -1 && m+f >= 30) {
      grade = 'D';
      if(m+f >= 50 || r >= 50) grade = 'C';
      if(m+f >= 65) grade = 'B';
      if(m+f >= 80) grade = 'A';
    }
    printf("%c\n", grade);
  }
  return 0;
}

File: introduction/ITP1_7_A-Grading.js
process.stdin.resume();
process.stdin.setEncoding('utf8');
process.stdin.on('data', function(chunk) {
  var lines = chunk.split('\n');
  lines.some(function(line) {
    var arr = line.split(' ').map(Number);
    var m = arr[0], f = arr[1], r = arr[2];
    if(m+f+r == -3) return true;
    var grade = 'F';
    if(m != -1 && f != -1 && m+f >= 30) {
      grade = 'D';
      if(m+f >= 50 || r >= 50) grade = 'C';
      if(m+f >= 65) grade = 'B';
      if(m+f >= 80) grade = 'A';
    }
    console.log(grade);
  });
});
